runners/cli/src/main/kotlin/org/jetbrains/dokka/main.kt

```diff
@@ -11,8 +11,12 @@ import java.nio.file.Paths
 class GlobalArguments(args: Array<String>) : DokkaConfiguration {
 
     val parser = ArgParser("globalArguments", prefixStyle = ArgParser.OptionPrefixStyle.JVM)
+    private val extraArgs: List<String> by parser.argument(ArgType.String, "Json file name or extra options", ).vararg().optional()
```

Comment:
`extraOptions` have a view of CLI arguments. Using them directly can be more convenient than encoding them into options.

Reply:
But options and arguments are not the same thing. Consider the following use cases for when you want to add extra options "one", "two", "three" and "four":

`-java -jar dokka-cli.jar -sourceSetName name -offlineMode one two three four`

vs

`java -jar dokka-cli.jar -sourceSet name name -offlineMode -extraOptions one,two,three,four`

Looking at the first example, I cannot deduce that these are _options_ - they look like some input/otuput arguments to me. In the second one it's clear

Using one argument for two things, especially a vararg, can also be very error-prone and confusing, and closes the door on ever adding a second argument. 

For instance, if you mistype some basic Dokka option, it will be silently parsed as an extra option instead of saying that such option does not exist.

Or if the provided json path is not correct:

previously:
```
ignat@workstation ~ [1]> java -jar dokka-cli-1.7.20.jar filepath.json
Exception in thread "main" java.io.FileNotFoundException: filepath.json (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at kotlin.io.FilesKt__FileReadWriteKt.readText(FileReadWrite.kt:125)
        at kotlin.io.FilesKt__FileReadWriteKt.readText$default(FileReadWrite.kt:125)
        at org.jetbrains.dokka.MainKt.initializeConfiguration(main.kt:15)
        at org.jetbrains.dokka.MainKt.main(main.kt:9
```

after this change:
```
ignat@workstation ~ [1]> java -jar dokka-cli-1.8.0-extra-options-SNAPSHOT.jar filepath.json
PROGRESS: Initializing plugins
Loaded plugins: []
Loaded: [

]
Suppressed: [

]
PROGRESS: 
Exception in thread "main" java.lang.IllegalStateException: ExtensionPoint: org.jetbrains.dokka.CoreExtensions/generation was expected to have exactly one extension registered, but none was found.
        at org.jetbrains.dokka.plugability.DokkaContextConfigurationImpl$single$1.invoke(DokkaContext.kt:151)
        at org.jetbrains.dokka.plugability.DokkaContextConfigurationImpl.single(DokkaContext.kt:158)
        at org.jetbrains.dokka.DokkaGenerator$generate$1.invoke(DokkaGenerator.kt:27)
        at org.jetbrains.dokka.DokkaGenerator$generate$1.invoke(DokkaGenerator.kt:17)
        at org.jetbrains.dokka.DokkaGeneratorKt.timed(DokkaGenerator.kt:76)
        at org.jetbrains.dokka.DokkaGeneratorKt.access$timed(DokkaGenerator.kt:1)
        at org.jetbrains.dokka.DokkaGenerator.generate(DokkaGenerator.kt:22)
        at org.jetbrains.dokka.MainKt.main(main.kt:431)
```

Reply:
> extra options "one", "two", "three" and "four":

They do not look like extra CLI options. In contrast, `-extraOptions -one,-two-,three,-four` can look weird without preprocessing. 

> For instance, if you mistype some basic Dokka option, it will be silently parsed as an extra option instead of saying that such option does not exist.

I agree It's a problem. Unfortunately, we do not check extra options at all.

> Using one argument for two things, especially a vararg, can also be very error-prone and confusing, and closes the door on ever adding a second argument.
> Or if the provided json path is not correct:

It is solvable.
 
 Anyway, we can discuss an option format and cases of using options.